cabal-install/Distribution/Client/Setup.hs

```diff
@@ -2416,8 +2416,7 @@ sdistCommand = Cabal.sdistCommand {
          (choiceOpt
             [ (Flag TargzFormat, ([], ["targz"]),
```

Comment:
I think we can drop the `--targz` flag altogether; it serves no real purpose anymore.

@23Skidoo related to the `data ArchiveFormat` type removal question; shall we drop the now redundant `--targz` flag too?